File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        long s=1;
        long e=x;
        long ans=0;
        while(s<=e)
        {
            long mid=(s+e)/2;
            long sq=mid*mid;
            if(sq==x)
            {
                return (int)mid;
            }
            else if(sq<x)
            {
                ans=mid;
                s=mid+1;
            }
            else
            {
                e=mid-1;
            }
        }
        return (int)ans;
    }
}

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        s=0
        e=x
        ans=0
        while(s<=e):
            mid=(s+e)//2
            sq=mid*mid
            if(sq==x):
                return mid
            elif(sq<x):
                ans=mid
                s=mid+1
            else:
                e=mid-1
        return ans